NetworkPkg/WifiConnectionManagerDxe/WifiConnectionMgrImpl.c

```diff
@@ -1489,6 +1489,12 @@ WifiMgrOnTimerTick (
   }
 
   Nic = (WIFI_MGR_DEVICE_DATA *)Context;
+  if (Nic->ConnectPendingNetwork == NULL) {
+    DEBUG ((DEBUG_INFO, "[WiFi Connection Manager] No profile for connection, no scan triggered!\n"));

```

Comment:
Should this be INFO or VERBOSE?

When is the timer canceled when a connection to a specific AP is successfull?

Reply:
moved to VERBOSE.

Added a CancelTimer in WifiMgrOnConnectFinished after connection has been validated as established.

Reply:
Ok.  So the general algorithm is to active the timer when user enters HII form to select an AP.  And the timer is canceled if either an AP connection fails or an AP connection succeeds.  What does the user have to do to perform a rescan?  Exit and enter the form?

Reply:
Yes, or since the set timer active is on a connect action, they just need to request a connection and the process will restart.